shared/sliceutil/slice.go

```diff
@@ -25,41 +25,83 @@ func SubsetUint64(a []uint64, b []uint64) bool {
 	return true
 }
 
-// IntersectionUint64 of two uint64 slices with time
+// IntersectionUint64 of any number of uint64 slices with time
 // complexity of approximately O(n) leveraging a map to
 // check for element existence off by a constant factor
 // of underlying map efficiency.
-func IntersectionUint64(a []uint64, b []uint64) []uint64 {
+func IntersectionUint64(s ...[]uint64) []uint64 {
+	if len(s) == 0 {
+		return []uint64{}
+	}
+	if len(s) == 1 {
+		return s[1]
```

Comment:
```suggestion
		return s[0]
```
shouldn't it be the 0 element?